terraform/setup/backend.tf

```diff
@@ -0,0 +1,65 @@
+# ------------------------------------------------------------------------
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+# -------------------------------------------------------------------------
+
+#In this module, we are uploading setup's terraform state to s3 bucket for two reasons:
+#-Share the setup configuration with others dev
+#-Avoid creating duplicate resources when sharing the same account
+
+#terraform {
```

Comment:
What are the comments for?

Reply:
The purpose of terraform backend s3 is to `uploading  terraform.tfstate to s3.` However, since the backend is not supporting condition checking with S3 bucket and DynamoDB Table currently; so first time dev setup which does not have DynamoDB Table and S3 bucket on their aws account will not able to pass the error. Therefore, it would be good for us to treat this as optional feature to upload it and let's the dev decide to use it or not. Also I have documented it out for everyone to know more about it: https://github.com/aws-observability/aws-otel-test-framework/pull/459/files#diff-0776469b54a50b9aaf826b5d5b9d11f4f34df0af615498fe9222343fdd9d8d33R26-R35

Reply:
Maybe this can be added to setup?

Reply:
Currently, it is in the setup.